go/vt/vtgate/buffer/shard_buffer.go

```diff
@@ -569,10 +569,32 @@ func (sb *shardBuffer) drain(q []*entry, err error) {
 	sb.timeoutThread.stop()
 
 	start := sb.timeNow()
-	// TODO(mberlin): Parallelize the drain by pumping the data through a channel.
+
+	// Parallelize the drain by pumping the data through a channel.
+	entryChan := make(chan *entry, len(q))
```

Comment:
Do I need to make the channel buffered? 🤔 I think it should be fine to not have it buffered, but I'm not 100% sure.

Reply:
I think this should not be buffered, definitely not buffered as `len(q)` because that's a lot of underlying memory being allocated!